pycbc/distributions/constraints.py

```diff
@@ -31,9 +35,24 @@ def __init__(self, constraint_arg, transforms=None, **kwargs):
         for kwarg in kwargs.keys():
             setattr(self, kwarg, kwargs[kwarg])
 
+        # For supernovae waveforms:
+        if 'principal_components_file' in kwargs:
+            pc_filename = kwargs['principal_components_file']
+            hull_dimention = numpy.array(kwargs['hull_dimention'])
+            self.hull_dimention = int(hull_dimention)
+            pc_file = h5py.File(pc_filename, 'r')
+            pc_coefficients = numpy.array(pc_file.get('coefficients'))
+            pc_file.close()
+            hull_points = []
+            for dim in range(self.hull_dimention):
+                hull_points.append(pc_coefficients[:, dim])
+            hull_points = numpy.array(hull_points).T
+            pc_coeffs_hull = Delaunay(hull_points)
+            self._hull = pc_coeffs_hull
+
```

Comment:
I'd rather not change the base Constraint class. These things are `ConvexHall` specific. So, add an `__init__` function to `ConvexHull`, and put this block of code there. See comment below for more details.